api-mock/api-mock-server.ts

```diff
@@ -111,68 +285,127 @@ app.use((req, res, next) => {
 });
 app.use(middleware());
 app.use(errorHandler());
-app.use((err, req, res, next) => {
-    log(err);
-    res.status(500).send("error");
-});
 
 app.get("/mock/ping", async (req, res, next) => {
     res.json({ ok: true });
 });
 
-app.post("/mock/reset", async (req, res, next) => {
-    Object.keys(vars).forEach((key) => (vars[key] = ""));
-    initST(req.body);
+app.post("/mock/init", async (req, res, next) => {
+    initST(req.body.config);
     res.json({ ok: true });
 });
 
-app.get("/mock/getMockedValues", async (req, res, next) => {
+app.get("/mock/mockedvalues", async (req, res, next) => {
+    let sessionVars = getSessionVars();
+    const vars: MockedVars = {
+        sendEmailToUserId,
+        token,
+        userPostPasswordReset,
+        emailPostPasswordReset,
+        sendEmailCallbackCalled,
+        sendEmailToUserEmail,
+        sendEmailToRecipeUserId,
+        userInCallback,
+        email,
+        newAccountInfoInCallback,
+        primaryUserInCallback: primaryUserInCallback?.toJson(),
+        userIdInCallback: userIdInCallback ?? sessionVars.userIdInCallback,
+        recipeUserIdInCallback:
+            recipeUserIdInCallback?.getAsString() ??
+            recipeUserIdInCallback ??
+            sessionVars.recipeUserIdInCallback?.getAsString(),
+        info,
+        store,
+    };
     res.json(vars);
 });
 
-app.post("/mock/EmailPassword/signup", async (req, res, next) => {
+app.post("/mock/setmockedvalues", async (req, res, next) => {
+    store = {
+        ...store,
+        ...req.body.store,
+    };
+    res.json({ ok: true });
+});
+
+app.post("/mock/resetmockedvalues", async (req, res, next) => {
+    resetVars();
+    res.json({ ok: true });
+});
+
+app.post("/mock/mockexternalapi", async (req, res, next) => {
+    const { url, status, body, path, method } = req.body;
+    nock(url)[method](path).reply(status, body);
+    res.json({ ok: true });
+});
+
+app.get("/mock/waitforevent", async (req, res, next) => {
     try {
-        log("EmailPassword:signup %j", req.body);
-        const response = await EmailPassword.signUp(
-            req.body.tenantId || constrains.tenantId,
-            req.body.email,
-            req.body.password,
-            req.body.session,
-            req.body.userContext
-        );
-        res.json(response);
+        log("ProcessState:waitForEvent %j", req.query);
+        const instance = ProcessState.getInstance();
+        const eventEnum = req.query.event ? Number(req.query.event) : null;
```

Comment:
why are we allowing this to be undefined here?

Reply:
is just type checking, instead of use req.query.event as string